src/pages/ReimbursementAccount/ReimbursementAccountPage.tsx

```diff
@@ -166,73 +192,41 @@ function ReimbursementAccountPage({route, policy}: ReimbursementAccountPageProps
-     */
-    const [hasACHDataBeenLoaded, setHasACHDataBeenLoaded] = useState(reimbursementAccount !== CONST.REIMBURSEMENT_ACCOUNT.DEFAULT_DATA);
-
-    const [shouldShowContinueSetupButton, setShouldShowContinueSetupButton] = useState(hasACHDataBeenLoaded ? getShouldShowContinueSetupButtonInitialValue() : false);
-    const [isReimbursementAccountLoading, setIsReimbursementAccountLoading] = useState(() => {
-        // By default return true (loading), if there are already loaded data we can skip the loading state
-        if (hasACHDataBeenLoaded && typeof reimbursementAccount?.isLoading === 'boolean' && !reimbursementAccount?.isLoading) {
-            return false;
-        }
-        return true;
-    });
-
-    // eslint-disable-next-line  @typescript-eslint/prefer-nullish-coalescing
-    const currentStep = achData?.currentStep || CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
-    const policyName = policy?.name ?? '';
-    const policyIDParam = route.params?.policyID ?? '-1';
-    const styles = useThemeStyles();
-    const {translate} = useLocalize();
-    const {isOffline} = useNetwork();
-    const requestorStepRef = useRef(null);
-    const prevIsReimbursementAccountLoading = usePrevious(reimbursementAccount?.isLoading);
-    const prevReimbursementAccount = usePrevious(reimbursementAccount);
-    const prevIsOffline = usePrevious(isOffline);
-
     /**
      * Retrieve verified business bank account currently being set up.
-     * @param ignoreLocalCurrentStep Pass true if you want the last "updated" view (from db), not the last "viewed" view (from onyx).
-     * @param ignoreLocalSubStep Pass true if you want the last "updated" view (from db), not the last "viewed" view (from onyx).
      */
-    function fetchData(ignoreLocalCurrentStep?: boolean, ignoreLocalSubStep?: boolean) {
-        // Show loader right away, as optimisticData might be set only later in case multiple calls are in the queue
-        BankAccounts.setReimbursementAccountLoading(true);
-
+    function fetchData() {
         // We can specify a step to navigate to by using route params when the component mounts.
         // We want to use the same stepToOpen variable when the network state changes because we can be redirected to a different step when the account refreshes.
         const stepToOpen = getStepToOpenFromRouteParams(route);
-        const subStep = achData?.subStep ?? '';
-        const localCurrentStep = achData?.currentStep ?? '';
-        BankAccounts.openReimbursementAccountPage(stepToOpen, ignoreLocalSubStep ? '' : subStep, ignoreLocalCurrentStep ? '' : localCurrentStep, policyIDParam);
+        const subStep = isPreviousPolicy ? achData?.subStep ?? '' : '';
+        const localCurrentStep = isPreviousPolicy ? achData?.currentStep ?? '' : '';
+        BankAccounts.openReimbursementAccountPage(stepToOpen, subStep, localCurrentStep, policyIDParam);
     }
 
     useEffect(() => {
-        if (!isReimbursementAccountLoading) {
```

Comment:
@koko57 Why do you remove this check?

Reply:
we no longer use isReimbursementAccountLoading here. I removed the local state we're just taking the info about the loading from Onyx. We don't need it locally

Reply:
Night owl 😄

Reply:
@koko57 I mean should we replace with `reimbursementAccount?.isLoading`

Reply:
temporarily different timezone 😃

Reply:
@koko57 Bump on this

Reply:
but here I'm changing the logic a bit. This runs only when component mounts. As we no longer set the state like this:

`    const [isReimbursementAccountLoading, setIsReimbursementAccountLoading] = useState(() => {
        // By default return true (loading), if there are already loaded data we can skip the loading state
        if (hasACHDataBeenLoaded && typeof reimbursementAccount?.isLoading === 'boolean' && !reimbursementAccount?.isLoading) {
            return false;
        }
        return true;
    });
`
we don't need to check if it's true or false and skip the content of the hook. We also don't need to check for reimbursementAccount?.isLoading, we set it true later and we actually want the whole hook to run on mount. We only don't want to fetch the data if it's the same policy as we already have the data.

Reply:
Got it

Reply:
Got it